e2e/fixtures/authenticatedUserPages.ts

```diff
@@ -0,0 +1,141 @@
+// ...based on: https://github.com/microsoft/playwright/issues/14570
+function createUserPageFixture<Page>(userPrefix: UserPrefixes) {
+  return async (
+    { browser, storageStatePaths },
+    use: (r: Page) => Promise<void>,
+  ) => {
+    const context = await browser.newContext({
+      storageState: storageStatePaths[userPrefix],
+    })
+    const validatedUserPage = await context.newPage()
+    await use(validatedUserPage)
+    await context.close()
+  }
+}
+
+export const testAuth = base.extend<
+  AuthenticatedUserPageTestFixtures,
+  AuthenticatedUserPageWorkerFixtures
+>({
+  // creates and deletes all test users defined in userConfig
+  // ...per worker + browserName
+  createUsers: [
+    async ({ browser }, use) => {
+      let userIds: string[] = []
+
+      const page = await base.step('setup page', async () => {
+        return await createPage(browser)
+      })
+
+      for (const user of Object.values(userConfigs)) {
+        await base.step(`create test user - ${user.username}`, async () => {
+          const userId = await createTestUser(user, getAdminPAT(), page)
+          expect(userId).not.toBeUndefined()
+          userIds.push(userId)
+        })
+      }
+
+      // close page so it isn't open while running all tests that depend on this worker fixture
+      await base.step('close page', async () => {
+        await page.close()
+      })
+
+      await use()
+
+      const cleanupPage = await createPage(browser)
+      for (const userId of userIds) {
+        await cleanupTestUser(userId, cleanupPage)
+      }
+      await cleanupPage.close()
+    },
+    // specify timeout so worker fixture duration doesn't skew test duration
+    // ...see https://github.com/microsoft/playwright/issues/15338
+    { scope: 'worker', timeout: 2 * 60 * 1000 },
+  ],
+  // stores auth state for each user in userConfig in storageState file, then deletes file
+  // ...per worker + browserName
+  // ...depends on createUsers worker fixture to create each test user
+  storageStatePaths: [
+    async ({ browser, browserName, createUsers }, use) => {
+      let storageStatePaths: StorageStatePaths = {}
+
+      for (const [userPrefix, user] of Object.entries(userConfigs)) {
+        const userPage = await base.step(
+          `create page: ${user.username}`,
+          async () => {
+            return await createPage(browser)
+          },
+        )
+
+        await base.step(`authenticate: ${user.username}`, async () => {
+          await loginTestUser(userPage, user.username, user.password)
+
+          // ensure that session_marker, i.e. user id, is set before saving state
+          await expect(async () => {
+            await getUserIdFromLocalStorage(userPage)
+          }).toPass()
+        })
+
+        await base.step(`save storageState: ${user.username}`, async () => {
+          const storageStateFileName = `${browserName}_${process.env.TEST_PARALLEL_INDEX}_${user.username}.json`
```

Comment:
Worker fixtures are setup/torn down per browser + worker, so include those identifiers in the storageState file for clarity when testing locally.